### ISSUE TITLE
metric parameter for KDTree 

### ISSUE THREAD

**Issue by pulkitmaloo** [scipy/scipy#8427]
Please provide a parameter for passing a metric in **scipy/scipy/spatial/kdtree.py**

**Comment by rgommers** [comment]
@pulkitmaloo a few more details (where exactly, motivation) would be useful. 

Also note that the `KDTree` implementation will be removed at some point, and the name `KDTree` will become an alias for `cKDTree`

**Comment by jakevdp** [comment]
Just FYI, if you're looking for nearest neighbors trees with flexible metrics, the scikit-learn [KDTree](http://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KDTree.html) and [BallTree](http://scikit-learn.org/stable/modules/generated/sklearn.neighbors.BallTree.html) accept a variety of distance metrics.

As far as adapting this kind of metric flexibility into scipy's cKDTree code, it would be a pretty significant amount of work, because the current code is very low-level and highly-optimized around assumptions made by the standard Euclidean metric.

**Comment by rgommers** [comment]
Thanks @jakevdp, forgot about that. That's a good solution, doesn't make sense to implement in SciPy then. Hence closing this enhancement request.
